core-services/service_account.tf

```diff
@@ -17,14 +19,19 @@ resource "kubernetes_cluster_role_binding" "massdriver-cloud-provisioner" {
   }
   subject {
     kind      = "ServiceAccount"
-    name      = kubernetes_service_account.massdriver-cloud-provisioner.metadata.0.name
-    namespace = kubernetes_service_account.massdriver-cloud-provisioner.metadata.0.namespace
+    name      = kubernetes_service_account_v1.massdriver-cloud-provisioner.metadata.0.name
+    namespace = kubernetes_service_account_v1.massdriver-cloud-provisioner.metadata.0.namespace
   }
 }
 
-data "kubernetes_secret" "massdriver-cloud-provisioner_service-account_secret" {
+resource "kubernetes_secret_v1" "massdriver-cloud-provisioner_token" {
```

Comment:
This is now a resource instead of a data since we have to create the secret as of 1.24, but this change is backwards compatible.  It *will* reset credentials for all the clusters, but that should only matter for local `kubeconfig` files.